src/native/clr/host/internal-pinvokes.cc

```diff
@@ -88,3 +90,105 @@ _monodroid_lookup_replacement_method_info (const char *jniSourceType, const char
 {
 	return JniRemapping::lookup_replacement_method_info (jniSourceType, jniMethodName, jniMethodSignature);
 }
+
+managed_timing_sequence* monodroid_timing_start (const char *message)
+{
+	// Technically a reference here is against the idea of shared pointers, but
+	// in this instance it's fine since we know we won't be storing the pointer
+	// and this way things are slightly faster.
+	std::shared_ptr<Timing> const &timing = Host::get_timing ();
```

Comment:
@grendello 
Should we use the same approach for JavaToManaged and ManagedToJava typemap lookup timings?
E.g.: https://github.com/dotnet/android/blob/bc9f07cf0c889b610b87585f2b21598566863ae8/src/native/clr/host/typemap.cc#L215
I could then add it to the other PR so it is more complete: https://github.com/dotnet/android/pull/9961

Reply:
The code in `typemap.cc` should no longer be necessary - we create the Timing instance early enough for it to be present when the first type mapping is requested